woocommerce/woocommerce
# Add phpcs-changed

### All Submissions:






### Changes proposed in this Pull Request:



This PR implements a wrapper that modifies PHPCS to pick up only changed lines of code instead of sniffing the entire file. This prevents sniff violations that the developer did not create and thus not obligated to fix. This will work as a pre-commit locally. Ref: https://github.com/sirbrillig/phpcs-changed

For CI implementation, we have opted to use a 3rd party marketplace action script ref: https://github.com/marketplace/actions/action-phpcs. This will retain the annotations that we currently have inline with the code but without linting the entire file but just the lines of code that was changed in the PR.

### How to test the changes in this Pull Request:

1. Create a branch off of this PR.
2. Run `pnpm install` then `pnpm -- turbo run build`
3. Add a hook `do_action( 'hook' );` to `plugins/woocommerce/includes/class-wc-ajax.php` (File that has previous un-documented hooks).
4. Run `git commit -am "Add new hook"`.
5. You should see a code sniff violation error on just the line of code you've added.
6. Now run `git commit --no-verify -am "Add new hook"` to bypass the validation.
7. Run `git push origin -u <BRANCH>`.
8. Create the PR on GitHub and let the CI run.
9. Ensure you see that the code sniff CI fails and you see in the details that the error is only on the line of code you've changed.

### Other information:




### FOR PR REVIEWER ONLY:
